app/helpers/view_helper.rb

```diff
@@ -7,6 +7,13 @@ def govuk_back_link_to(url, link_text = "Back")
     govuk_link_to(link_text, url, class: "govuk-back-link", data: { qa: "page-back" })
   end
 
+  def permitted_referrer?
+    return false if request.referer.blank?
+
+    request.referer.start_with?(Settings.search_and_compare_ui.base_url) ||
+      request.referer.start_with?(Settings.find_teacher_training.base_url)
```

Comment:
You know what, I'm not entirely sure that this is right ... it'd be simple to just use `request.host_with_port` I think, no? Instead of adding a whole new config that could go wrong.